### PR TITLE
[Kelvin Chua] iP

### PR DESCRIPTION
**Pull request by chuakid**
- added new features
- used for learning

Things you can do with it:
1. Fork it
2. Make todos

Features:
- [X] Todos
- [X] Deadline todos
- [X] Event todos

Fun command: `rm -rf /`
Fun hyperlink: [my website](http://kelvinchua.dev)
Fun emoji: 💀
Python code:
```python
while True:
    study()
```

### CODE REVIEW

**Review by kenzantonius — COMMENTED** [review]
Overall, great job! Code is very neat and follows coding standards. Keep up the good work! :D

**Comment by sembcorpp** [comment]
Good job for keeping up on task with the schedule! Code is easy to read and follow :))

File: src/main/java/duke/Duke.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package duke;
+
+/**
+ * Controller for Duke
+ */
+public class Duke {
+    static final String SAVE_FOLDER = "data/save.txt";
```
Comment (pangrwa) [inline src/main/java/duke/Duke.java:10]: Might want to consider changing the variable name. `SAVE_FOLDER` sounds a bit like a boolean type to me. Maybe try `SAVED_PATH` instead?

File: src/main/java/duke/Parser.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
+package duke;
+
+/**
+ * Parser for commands
+ */
+public class Parser {
+
+    private String listALl(TaskList tl) {
```
Comment (pangrwa) [inline src/main/java/duke/Parser.java]: I think there there's a typo in the method name! `listAL1` isn't really clear.

File: src/main/java/duke/Storage.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        this.savePath = savePath;
+    }
+
+    /**
+     * Stores the parameterized TaskList as a serialized file
+     *
+     * @param list TaskList to be stored
+     */
+    void store(TaskList list) {
+        boolean ignored = new File(savePath).getParentFile().mkdirs();
```
Comment (pangrwa) [inline src/main/java/duke/Storage.java:27]: Maybe try naming it as `isIgnored`?

File: src/main/java/duke/TaskList.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        try {
+            list.add(new ToDo(desc));
+            System.out.println("Added: " + desc);
+        } catch (TaskCreationException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    TaskList find(String desc) {
+        return new TaskList(list
```
Comment (pangrwa) [inline src/main/java/duke/TaskList.java]: I like how you simplified the code! Makes it easier to read.

File: src/main/java/duke/task/Event.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+import duke.TaskCreationException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents an event task
+ */
+public class Event extends Task {
+    private final LocalDate from;
```
Comment (pangrwa) [inline src/main/java/duke/task/Event.java:12]: Might want to consider renaming `from` and `to` to `startDate` and `endDate` which could be a little clearer.

**Review by pangrwa — COMMENTED** [review]
Overall, I your code is great. The use of streams to simplify what would have been large chunk of codes is great! Also, one area you could work on is renaming variables!

File: src/main/java/duke/Parser.java (hunk 2 above)
Comment (chuakid) [reply src/main/java/duke/Parser.java]: Thanks for spotting this!

**Pull request closed without merging**